hat/app/org/hatdex/hat/she/service/FunctionService.scala

```diff
@@ -134,18 +134,24 @@ class FunctionService @Inject() (
       .map(_.get)
   }
 
-  def run(configuration: FunctionConfiguration, startTime: Option[DateTime])(implicit hatServer: HatServer): Future[Done] = {
+  def run(configuration: FunctionConfiguration, startTime: Option[DateTime], useAll: Boolean)(implicit hatServer: HatServer): Future[Done] = {
     val executionTime = DateTime.now()
     logger.info(s"[${hatServer.domain}] SHE function [${configuration.id}] run @$executionTime (previously $startTime)")
     functionRegistry.get[FunctionExecutable](configuration.id)
       .map { function: FunctionExecutable =>
         val fromDate = startTime.orElse(Some(DateTime.now().minusMonths(6)))
         val untilDate = Some(DateTime.now().plusMonths(3))
+        val dataBundleTimeFilter = if (useAll) {
+          None
+        }
+        else {
+          configuration.status.lastExecution
+        }
+
         val executionResult = for {
           _ ← markExecuting(configuration)
           bundle ← function.bundleFilterByDate(fromDate, untilDate)
-          data ← dataService.bundleData(bundle, createdAfter = configuration.status.lastExecution) // Get all bundle data from a specific date until now
-          // data ← dataService.bundleData(bundle, createdAfter = None) // TODO: Terry's comment: Setting createdAfter to None will allow the use case of - Show me the word distribution of my last 100 tweets.
+          data ← dataService.bundleData(bundle, createdAfter = dataBundleTimeFilter) // Get all bundle data from a specific date until now
```

Comment:
i dont know what our strategy is for removing unicode characters ?

Reply:
I saw that as well but I decided -> 🤐 since I don't think it's a broken window 🤔

Reply:
unfortunately unicode characters are becoming deprecated in 2.13 so we should start removing them [https://github.com/scala/scala/pull/7540/files]

Reply:
It is a broken window, but would've assumed that it can be done in one go using one of the code formatting libraries.

Reply:
In .scalafmt.conf

rewriteTokens {
  "⇒" = "=>"
  "→" = "->"
  "←" = "<-"
}

Reply:
I think that's the first rule to add for the formatting.

Reply:
agree lets do formatting as separate PR's from logic changes

Reply:
Agreed, HAT project has additional complication that Scalariform needs to be removed first